examples/azure/functions.ipynb

```diff
@@ -0,0 +1,431 @@
+    "\n",
+    "**NOTE**: Chat functions require model versions beginning with gpt-4 and gpt-35-turbo's `-0613` labels. They are not supported by older versions of the models."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup\n",
+    "\n",
+    "First, we install the necessary dependencies."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! pip install openai\n",
+    "# (Optional) If you want to use Microsoft Active Directory\n",
+    "! pip install azure-identity"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import openai"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "Additionally, to properly access the Azure OpenAI Service, we need to create the proper resources at the [Azure Portal](https://portal.azure.com) (you can check a detailed guide on how to do this in the [Microsoft Docs](https://learn.microsoft.com/en-us/azure/cognitive-services/openai/how-to/create-resource?pivots=web-portal))\n",
+    "\n",
+    "Once the resource is created, the first thing we need to use is its endpoint. You can get the endpoint by looking at the *\"Keys and Endpoints\"* section under the *\"Resource Management\"* section. Having this, we will set up the SDK using this information:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "openai.api_base = \"\" # Add your endpoint here\n",
+    "\n",
+    "# functions is only supported by the 2023-07-01-preview API version\n",
+    "openai.api_version = \"2023-07-01-preview\""
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Authentication\n",
+    "\n",
+    "The Azure OpenAI service supports multiple authentication mechanisms that include API keys and Azure credentials."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "use_azure_active_directory = False"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "#### Authentication using API key\n",
+    "\n",
+    "To set up the OpenAI SDK to use an *Azure API Key*, we need to set up the `api_type` to `azure` and set `api_key` to a key associated with your endpoint (you can find this key in *\"Keys and Endpoints\"* under *\"Resource Management\"* in the [Azure Portal](https://portal.azure.com))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if not use_azure_active_directory:\n",
+    "    openai.api_type = \"azure\"\n",
+    "    openai.api_key = \"\"  # Add your api key here"
```

Comment:
Happy to merge.

One thing I noticed here is that we're encouraging developers to put their API keys in code. In general, I really want to discourage this, as the leak of an API key can be quite costly. Do you mind removing this and using environment variables instead?

Reply:
Thanks for reviewing, that's good feedback. I updated the code to use an environment variable for API key across all the azure examples.

Reply:
Awesome. Thank you!